### DISCUSSION TITLE
how to get the result immediately

### DISCUSSION THREAD

**Discussion by take-study** [OpenModelica/OpenModelica#12576] [General]
i want know is there a way to get the simulate result immediately. now there is the result file where i can get the simulate result,but the all scripting api for read data will wait the simulate over, so i want find a way to get the data when the simulate is still running?

**Comment by casella** [comment]
@sjoelund can we read the .mat file as it is being written?

**Reply by sjoelund** [reply]
Yes and no. On Windows, you cannot open a file for reading if it is being written to. On Linux, you can. But due to buffering you might not be able to read continuously, and the file header would be invalid since the number of rows is still unknown.

You can get results during simulation by reading via the OPC-UA interface (and I think some other things added by other developers).

**Reply by casella** [reply]
> On Windows, you cannot open a file for reading if it is being written to.

Are you sure? Dymola makes the results available as the executable is running. But maybe it does it in some other way than through the .mat file?

**Reply by casella** [reply]
In fact, for very long simulations, being able to inspect the results while they are being generated is a must-have feature.

Maybe we could close and reopen them every now and then, to give OMEdit a chance to read them?

**Reply by take-study** [reply]
> Yes and no. On Windows, you cannot open a file for reading if it is being written to. On Linux, you can. But due to buffering you might not be able to read continuously, and the file header would be invalid since the number of rows is still unknown.
> 
> You can get results during simulation by reading via the OPC-UA interface (and I think some other things added by other developers).

@sjoelund  in fact i am using the `omc  zmq model` for multi process too, so i want find a way that won't use more tcp port. i have noticed that there is a simulation runtime flag [--embeddedServer=value](https://openmodelica.org/doc/OpenModelicaUsersGuide/1.23/simulationflags.html#simflag-embeddedserver)  with the `filename` valid value, is that means i can use a file to replace the tcp connection to get the result when simulation?
